subsys/bluetooth/controller/ll_sw/ull_adv_aux.c

```diff
@@ -71,11 +71,18 @@ static void ticker_cb(uint32_t ticks_at_expire, uint32_t ticks_drift,
 static struct ll_adv_aux_set ll_adv_aux_pool[CONFIG_BT_CTLR_ADV_AUX_SET];
 static void *adv_aux_free;
 
-#if defined(CONFIG_BT_CTLR_ADV_PERIODIC) && defined(CONFIG_BT_TICKER_EXT_EXPIRE_INFO)
+#if defined(CONFIG_BT_CTLR_ADV_AUX_SLOT_WINDOW) || \
+	(defined(CONFIG_BT_CTLR_ADV_PERIODIC) && \
+	 defined(CONFIG_BT_TICKER_EXT_EXPIRE_INFO))
+#if defined(CONFIG_BT_CTLR_ADV_PERIODIC) && \
+	defined(CONFIG_BT_TICKER_EXT_EXPIRE_INFO)
```

Comment:
Man, this some tasty `defined` spaghetti :spaghetti: 

I assume that it required given the existing dependencies on `#if defined` in the controller, where nearly everything is guarded one way or another. 

At some point it would be interesting to start removing these defined for all prototypes (both static and the ones in header files), so we can start using `IS_ENABLED` more.

Reply:
Unfortunately, ISO feature scheduling requirements have been challenging, and it has been difficult to avoid regressions in behavior change without conditionally compiling out new implementations.